src/main/java/com/iota/iri/utils/log/interval/IntervalProgressLogger.java

```diff
@@ -0,0 +1,244 @@
+package com.iota.iri.utils.log.interval;
+
+import com.iota.iri.utils.log.ProgressLogger;
+import org.slf4j.Logger;
+
+import java.text.DecimalFormat;
+
+/**
+ * Implements the basic contract of the {@link ProgressLogger} while showing the progress as messages in the console.
+ *
+ * Instead of printing all messages immediately and unnecessarily spamming the console output, it rate limits the amount
+ * of messages shown and only prints updates in a pre-defined interval.
+ */
+public class IntervalProgressLogger implements ProgressLogger {
+    /**
+     * Holds the default interval in which the logger shows messages.
+     */
+    private static final int DEFAULT_LOG_INTERVAL = 3000;
```

Comment:
This is missing a comment detailing the temporal resolution of this inteval value (or, suffixing with _MS or similar)